### ISSUE TITLE
BUG: Unable to build scipy due to import in f2py

### ISSUE THREAD

**Issue by JaRoSchm** [scipy/scipy#21527]
### Describe your issue.

Hi, I just tried to build scipy locally following this guide: https://docs.scipy.org/doc/scipy/dev/dev_quickstart.html. However the command `python dev.py build` fails. It seems that the standard library `signal` is mixed up with `scipy.signal`.

### Reproducing Code Example

```python
python dev.py build
```

### Error message

```shell
💻  meson setup /home/jschmidt/Devel/scipy_perf/scipy/build --prefix 
/home/jschmidt/Devel/scipy_perf/scipy/build-install
The Meson build system
Version: 1.5.1
Source dir: /home/jschmidt/Devel/scipy_perf/scipy
Build dir: /home/jschmidt/Devel/scipy_perf/scipy/build
Build type: native build
Project name: scipy
Project version: 1.15.0.dev0+git20240908.47bc9b6
C compiler for the host machine: /home/jschmidt/mambaforge/envs/scipy-dev/bin/x86_64-conda-linux-gnu-cc (gcc 13.3.0 "x86_64-conda-linux-gnu-cc (conda-forge gcc 13.3.0-1) 13.3.0")
C linker for the host machine: /home/jschmidt/mambaforge/envs/scipy-dev/bin/x86_64-conda-linux-gnu-cc ld.bfd 2.40
C++ compiler for the host machine: /home/jschmidt/mambaforge/envs/scipy-dev/bin/x86_64-conda-linux-gnu-c++ (gcc 13.3.0 "x86_64-conda-linux-gnu-c++ (conda-forge gcc 13.3.0-1) 13.3.0")
C++ linker for the host machine: /home/jschmidt/mambaforge/envs/scipy-dev/bin/x86_64-conda-linux-gnu-c++ ld.bfd 2.40
Cython compiler for the host machine: cython (cython 3.0.11)
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program python3 found: YES (/home/jschmidt/mambaforge/envs/scipy-dev/bin/python3.12)
Found pkg-config: YES (/home/jschmidt/mambaforge/envs/scipy-dev/bin/pkg-config) 0.29.2
Run-time dependency python found: YES 3.12
Program cython found: YES (/home/jschmidt/mambaforge/envs/scipy-dev/bin/cython)
Compiler for C supports arguments -Wno-unused-but-set-variable: YES 
Compiler for C supports arguments -Wno-unused-function: YES 
Compiler for C supports arguments -Wno-conversion: YES 
Compiler for C supports arguments -Wno-misleading-indentation: YES 
Library m found: YES
Fortran compiler for the host machine: /home/jschmidt/mambaforge/envs/scipy-dev/bin/x86_64-conda-linux-gnu-gfortran (gcc 13.3.0 "GNU Fortran (conda-forge gcc 13.3.0-1) 13.3.0")
Fortran linker for the host machine: /home/jschmidt/mambaforge/envs/scipy-dev/bin/x86_64-conda-linux-gnu-gfortran ld.bfd 2.40
Compiler for Fortran supports arguments -Wno-conversion: YES 
Checking if "-Wl,--version-script" : links: YES 
Program tools/generate_f2pymod.py found: YES (/home/jschmidt/mambaforge/envs/scipy-dev/bin/python3.12 /home/jschmidt/Devel/scipy_perf/scipy/tools/generate_f2pymod.py)
Program scipy/_build_utils/tempita.py found: YES (/home/jschmidt/mambaforge/envs/scipy-dev/bin/python3.12 /home/jschmidt/Devel/scipy_perf/scipy/scipy/_build_utils/tempita.py)
Program pythran found: YES 0.16.1 0.16.1 (/home/jschmidt/mambaforge/envs/scipy-dev/bin/pythran)
Found CMake: /usr/bin/cmake (3.28.3)
WARNING: CMake Toolchain: Failed to determine CMake compilers state
Run-time dependency xsimd found: NO (tried pkgconfig and cmake)
Run-time dependency threads found: YES
Library npymath found: YES
Run-time dependency pybind11 found: YES 2.13.5
Program f2py found: YES (/home/jschmidt/mambaforge/envs/scipy-dev/bin/f2py)

scipy/meson.build:158:15: ERROR: Command `/home/jschmidt/mambaforge/envs/scipy-dev/bin/f2py -v` failed with status 1.

A full log can be found at /home/jschmidt/Devel/scipy_perf/scipy/build/meson-logs/meson-log.txt
Meson build setup failed!
```

More details from `meson-log.txt` (only last part, as the whole output is too long for GitHub):

```
...

Program f2py found: YES (/home/jschmidt/mambaforge/envs/scipy-dev/bin/f2py)
Running command: /home/jschmidt/mambaforge/envs/scipy-dev/bin/f2py -v
--- stdout ---

--- stderr ---
Traceback (most recent call last):
  File "/home/jschmidt/mambaforge/envs/scipy-dev/bin/f2py", line 7, in <module>
    from numpy.f2py.f2py2e import main
  File "/home/jschmidt/mambaforge/envs/scipy-dev/lib/python3.12/site-packages/numpy/f2py/__init__.py", line 14, in <module>
    import subprocess
  File "/home/jschmidt/mambaforge/envs/scipy-dev/lib/python3.12/subprocess.py", line 49, in <module>
    import signal
  File "/home/jschmidt/Devel/scipy_perf/scipy/scipy/signal/__init__.py", line 292, in <module>
    from . import _sigtools, windows
  File "/home/jschmidt/Devel/scipy_perf/scipy/scipy/signal/windows/__init__.py", line 42, in <module>
    from ._windows import *
  File "/home/jschmidt/Devel/scipy_perf/scipy/scipy/signal/windows/_windows.py", line 7, in <module>
    from scipy import linalg, special, fft as sp_fft
ModuleNotFoundError: No module named 'scipy'



scipy/meson.build:158:15: ERROR: Command `/home/jschmidt/mambaforge/envs/scipy-dev/bin/f2py -v` failed with status 1.
```

### SciPy/NumPy/Python version and system information

I'm using mamba 1.5.9 and a freshly build scipy-dev environment.

<details>

```
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
_sysroot_linux-64_curr_repodata_hack 3                   h69a702a_16    conda-forge
accessible-pygments       0.0.5              pyhd8ed1ab_0    conda-forge
alabaster                 0.7.16             pyhd8ed1ab_0    conda-forge
alsa-lib                  1.2.12               h4ab18f5_0    conda-forge
anyio                     4.4.0              pyhd8ed1ab_0    conda-forge
argon2-cffi               23.1.0             pyhd8ed1ab_0    conda-forge
argon2-cffi-bindings      21.2.0          py312h66e93f0_5    conda-forge
array-api-strict          2.0.1              pyhd8ed1ab_0    conda-forge
arrow                     1.3.0              pyhd8ed1ab_0    conda-forge
asttokens                 2.4.1              pyhd8ed1ab_0    conda-forge
asv                       0.6.4           py312h2ec8cdc_1    conda-forge
asv_runner                0.2.1              pyhd8ed1ab_0    conda-forge
attrs                     24.2.0             pyh71513ae_0    conda-forge
babel                     2.14.0             pyhd8ed1ab_0    conda-forge
backports.zoneinfo        0.2.1           py312h7900ff3_9    conda-forge
beautifulsoup4            4.12.3             pyha770c72_0    conda-forge
beniget                   0.4.2.post1        pyhd8ed1ab_0    conda-forge
binutils                  2.40                 h4852527_7    conda-forge
binutils_impl_linux-64    2.40                 ha1999f0_7    conda-forge
binutils_linux-64         2.40                 hb3c18ed_2    conda-forge
bleach                    6.1.0              pyhd8ed1ab_0    conda-forge
brotli                    1.1.0                hb9d3cd8_2    conda-forge
brotli-bin                1.1.0                hb9d3cd8_2    conda-forge
brotli-python             1.1.0           py312h2ec8cdc_2    conda-forge
bzip2                     1.0.8                h4bc722e_7    conda-forge
c-compiler                1.8.0                h2b85faf_0    conda-forge
ca-certificates           2024.8.30            hbcca054_0    conda-forge
cached-property           1.5.2                hd8ed1ab_1    conda-forge
cached_property           1.5.2              pyha770c72_1    conda-forge
cairo                     1.18.0               hebfffa5_3    conda-forge
certifi                   2024.8.30          pyhd8ed1ab_0    conda-forge
cffi                      1.17.1          py312h06ac9bb_0    conda-forge
charset-normalizer        3.3.2              pyhd8ed1ab_0    conda-forge
click                     8.1.7           unix_pyh707e725_0    conda-forge
cloudpickle               3.0.0              pyhd8ed1ab_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
colorlog                  6.8.2           py312h7900ff3_1    conda-forge
comm                      0.2.2              pyhd8ed1ab_0    conda-forge
compilers                 1.8.0                ha770c72_0    conda-forge
contourpy                 1.3.0           py312h68727a3_1    conda-forge
coverage                  7.6.1           py312h66e93f0_1    conda-forge
cxx-compiler              1.8.0                h1a2810e_0    conda-forge
cycler                    0.12.1             pyhd8ed1ab_0    conda-forge
cython                    3.0.11          py312h2ec8cdc_2    conda-forge
cython-lint               0.16.2             pyhd8ed1ab_0    conda-forge
dbus                      1.13.6               h5008d03_3    conda-forge
debugpy                   1.8.5           py312h2ec8cdc_1    conda-forge
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
distlib                   0.3.8              pyhd8ed1ab_0    conda-forge
docutils                  0.21.2             pyhd8ed1ab_0    conda-forge
doit                      0.36.0             pyhd8ed1ab_0    conda-forge
double-conversion         3.3.0                h59595ed_0    conda-forge
entrypoints               0.4                pyhd8ed1ab_0    conda-forge
exceptiongroup            1.2.2              pyhd8ed1ab_0    conda-forge
execnet                   2.1.1              pyhd8ed1ab_0    conda-forge
executing                 2.1.0              pyhd8ed1ab_0    conda-forge
expat                     2.6.3                h5888daf_0    conda-forge
filelock                  3.16.0             pyhd8ed1ab_0    conda-forge
font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge
font-ttf-inconsolata      3.000                h77eed37_0    conda-forge
font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge
font-ttf-ubuntu           0.83                 h77eed37_2    conda-forge
fontconfig                2.14.2               h14ed4e7_0    conda-forge
fonts-conda-ecosystem     1                             0    conda-forge
fonts-conda-forge         1                             0    conda-forge
fonttools                 4.53.1          py312h66e93f0_1    conda-forge
fortran-compiler          1.8.0                h36df796_0    conda-forge
fqdn                      1.5.1              pyhd8ed1ab_0    conda-forge
freetype                  2.12.1               h267a509_2    conda-forge
gast                      0.5.5              pyhd8ed1ab_0    conda-forge
gcc                       13.3.0               h9576a4e_1    conda-forge
gcc_impl_linux-64         13.3.0               hfea6d02_1    conda-forge
gcc_linux-64              13.3.0               hc28eda2_2    conda-forge
gfortran                  13.3.0               h9576a4e_1    conda-forge
gfortran_impl_linux-64    13.3.0               h10434e7_1    conda-forge
gfortran_linux-64         13.3.0               hb919d3a_2    conda-forge
gmp                       6.3.0                hac33072_2    conda-forge
gmpy2                     2.1.5           py312h7201bc8_2    conda-forge
graphite2                 1.3.13            h59595ed_1003    conda-forge
greenlet                  3.0.3           py312h30efb56_0    conda-forge
gxx                       13.3.0               h9576a4e_1    conda-forge
gxx_impl_linux-64         13.3.0               hdbfa832_1    conda-forge
gxx_linux-64              13.3.0               h6834431_2    conda-forge
h2                        4.1.0              pyhd8ed1ab_0    conda-forge
harfbuzz                  9.0.0                hda332d3_1    conda-forge
hpack                     4.0.0              pyh9f0ad1d_0    conda-forge
hyperframe                6.0.1              pyhd8ed1ab_0    conda-forge
hypothesis                6.112.0            pyha770c72_0    conda-forge
icu                       75.1                 he02047a_0    conda-forge
idna                      3.8                pyhd8ed1ab_0    conda-forge
imagesize                 1.4.1              pyhd8ed1ab_0    conda-forge
importlib-metadata        8.4.0              pyha770c72_0    conda-forge
importlib_metadata        8.4.0                hd8ed1ab_0    conda-forge
importlib_resources       6.4.4              pyhd8ed1ab_0    conda-forge
iniconfig                 2.0.0              pyhd8ed1ab_0    conda-forge
intersphinx-registry      0.2408.13          pyhd8ed1ab_0    conda-forge
ipykernel                 6.29.5             pyh3099207_0    conda-forge
ipython                   8.27.0             pyh707e725_0    conda-forge
isoduration               20.11.0            pyhd8ed1ab_0    conda-forge
jedi                      0.19.1             pyhd8ed1ab_0    conda-forge
jinja2                    3.1.4              pyhd8ed1ab_0    conda-forge
json5                     0.9.25             pyhd8ed1ab_0    conda-forge
jsonpointer               3.0.0           py312h7900ff3_1    conda-forge
jsonschema                4.23.0             pyhd8ed1ab_0    conda-forge
jsonschema-specifications 2023.12.1          pyhd8ed1ab_0    conda-forge
jsonschema-with-format-nongpl 4.23.0               hd8ed1ab_0    conda-forge
jupyter-cache             1.0.0              pyhd8ed1ab_0    conda-forge
jupyter_client            8.6.2              pyhd8ed1ab_0    conda-forge
jupyter_core              5.7.2           py312h7900ff3_0    conda-forge
jupyter_events            0.10.0             pyhd8ed1ab_0    conda-forge
jupyter_server            2.14.2             pyhd8ed1ab_0    conda-forge
jupyter_server_terminals  0.5.3              pyhd8ed1ab_0    conda-forge
jupyterlab_pygments       0.3.0              pyhd8ed1ab_1    conda-forge
jupyterlab_server         2.27.3             pyhd8ed1ab_0    conda-forge
jupyterlite-core          0.4.1              pyhd8ed1ab_0    conda-forge
jupyterlite-pyodide-kernel 0.4.2              pyhd8ed1ab_0    conda-forge
jupyterlite-sphinx        0.16.5             pyhd8ed1ab_0    conda-forge
jupytext                  1.16.4             pyh80e38bb_0    conda-forge
kernel-headers_linux-64   3.10.0              h4a8ded7_16    conda-forge
keyutils                  1.6.1                h166bdaf_0    conda-forge
kiwisolver                1.4.7           py312h68727a3_0    conda-forge
krb5                      1.21.3               h659f571_0    conda-forge
lcms2                     2.16                 hb7c19ff_0    conda-forge
ld_impl_linux-64          2.40                 hf3520f5_7    conda-forge
lerc                      4.0.0                h27087fc_0    conda-forge
libblas                   3.9.0           23_linux64_openblas    conda-forge
libbrotlicommon           1.1.0                hb9d3cd8_2    conda-forge
libbrotlidec              1.1.0                hb9d3cd8_2    conda-forge
libbrotlienc              1.1.0                hb9d3cd8_2    conda-forge
libcblas                  3.9.0           23_linux64_openblas    conda-forge
libclang-cpp18.1          18.1.8          default_hf981a13_4    conda-forge
libclang13                18.1.8          default_h9def88c_4    conda-forge
libcups                   2.3.3                h4637d8d_4    conda-forge
libdeflate                1.21                 h4bc722e_0    conda-forge
libdrm                    2.4.123              hb9d3cd8_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libegl                    1.7.0                ha4b6fd6_0    conda-forge
libexpat                  2.6.3                h5888daf_0    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc                    14.1.0               h77fa898_1    conda-forge
libgcc-devel_linux-64     13.3.0             h84ea5a7_101    conda-forge
libgcc-ng                 14.1.0               h69a702a_1    conda-forge
libgfortran               14.1.0               h69a702a_1    conda-forge
libgfortran-ng            14.1.0               h69a702a_1    conda-forge
libgfortran5              14.1.0               hc5f4f2c_1    conda-forge
libgl                     1.7.0                ha4b6fd6_0    conda-forge
libglib                   2.80.3               h315aac3_2    conda-forge
libglvnd                  1.7.0                ha4b6fd6_0    conda-forge
libglx                    1.7.0                ha4b6fd6_0    conda-forge
libgomp                   14.1.0               h77fa898_1    conda-forge
libiconv                  1.17                 hd590300_2    conda-forge
libjpeg-turbo             3.0.0                hd590300_1    conda-forge
liblapack                 3.9.0           23_linux64_openblas    conda-forge
libllvm18                 18.1.8               h8b73ec9_2    conda-forge
libnsl                    2.0.1                hd590300_0    conda-forge
libopenblas               0.3.27          pthreads_hac2b453_1    conda-forge
libpciaccess              0.18                 hd590300_0    conda-forge
libpng                    1.6.43               h2797004_0    conda-forge
libpq                     16.4                 h2d7952a_1    conda-forge
libsanitizer              13.3.0               heb74ff8_1    conda-forge
libsodium                 1.0.20               h4ab18f5_0    conda-forge
libsqlite                 3.46.1               hadc24fc_0    conda-forge
libstdcxx                 14.1.0               hc0a3c3a_1    conda-forge
libstdcxx-devel_linux-64  13.3.0             h84ea5a7_101    conda-forge
libstdcxx-ng              14.1.0               h4852527_1    conda-forge
libtiff                   4.6.0                h46a8edc_4    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libwebp-base              1.4.0                hd590300_0    conda-forge
libxcb                    1.16                 hb9d3cd8_1    conda-forge
libxcrypt                 4.4.36               hd590300_1    conda-forge
libxkbcommon              1.7.0                h2c5496b_1    conda-forge
libxml2                   2.12.7               he7c6b58_4    conda-forge
libxslt                   1.1.39               h76b75d6_0    conda-forge
libzlib                   1.3.1                h4ab18f5_1    conda-forge
markdown-it-py            3.0.0              pyhd8ed1ab_0    conda-forge
markupsafe                2.1.5           py312h66e93f0_1    conda-forge
matplotlib                3.9.2           py312h7900ff3_0    conda-forge
matplotlib-base           3.9.2           py312h854627b_0    conda-forge
matplotlib-inline         0.1.7              pyhd8ed1ab_0    conda-forge
mdit-py-plugins           0.4.1              pyhd8ed1ab_0    conda-forge
mdurl                     0.1.2              pyhd8ed1ab_0    conda-forge
meson                     1.5.1              pyhd8ed1ab_1    conda-forge
meson-python              0.16.0             pyh0c530f3_0    conda-forge
mistune                   3.0.2              pyhd8ed1ab_0    conda-forge
mpc                       1.3.1                h24ddda3_1    conda-forge
mpfr                      4.2.1                h90cbb55_3    conda-forge
mpmath                    1.3.0              pyhd8ed1ab_0    conda-forge
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
mypy                      1.11.2          py312h66e93f0_0    conda-forge
mypy_extensions           1.0.0              pyha770c72_0    conda-forge
mysql-common              9.0.1                h70512c7_0    conda-forge
mysql-libs                9.0.1                ha479ceb_0    conda-forge
myst-nb                   1.1.1              pyhd8ed1ab_0    conda-forge
myst-parser               4.0.0              pyhd8ed1ab_0    conda-forge
nbclient                  0.10.0             pyhd8ed1ab_0    conda-forge
nbconvert-core            7.16.4             pyhd8ed1ab_1    conda-forge
nbformat                  5.10.4             pyhd8ed1ab_0    conda-forge
ncurses                   6.5                  he02047a_1    conda-forge
nest-asyncio              1.6.0              pyhd8ed1ab_0    conda-forge
ninja                     1.12.1               h297d8ca_0    conda-forge
numpy                     2.1.1           py312h58c1407_0    conda-forge
numpydoc                  1.8.0              pyhd8ed1ab_0    conda-forge
openblas                  0.3.27          pthreads_h9eca1d5_1    conda-forge
openjpeg                  2.5.2                h488ebb8_0    conda-forge
openssl                   3.3.2                hb9d3cd8_0    conda-forge
overrides                 7.7.0              pyhd8ed1ab_0    conda-forge
packaging                 24.1               pyhd8ed1ab_0    conda-forge
pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
parso                     0.8.4              pyhd8ed1ab_0    conda-forge
pcre2                     10.44                hba22ea6_2    conda-forge
pexpect                   4.9.0              pyhd8ed1ab_0    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    10.4.0          py312h287a98d_0    conda-forge
pip                       24.2               pyh8b19718_1    conda-forge
pixman                    0.43.2               h59595ed_0    conda-forge
pkg-config                0.29.2            h4bc722e_1009    conda-forge
pkginfo                   1.11.1             pyhd8ed1ab_0    conda-forge
pkgutil-resolve-name      1.3.10             pyhd8ed1ab_1    conda-forge
platformdirs              4.3.2              pyhd8ed1ab_0    conda-forge
pluggy                    1.5.0              pyhd8ed1ab_0    conda-forge
ply                       3.11               pyhd8ed1ab_2    conda-forge
pooch                     1.8.2              pyhd8ed1ab_0    conda-forge
prometheus_client         0.20.0             pyhd8ed1ab_0    conda-forge
prompt-toolkit            3.0.47             pyha770c72_0    conda-forge
psutil                    6.0.0           py312h66e93f0_1    conda-forge
pthread-stubs             0.4               h36c2ea0_1001    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
pure_eval                 0.2.3              pyhd8ed1ab_0    conda-forge
pybind11                  2.13.5          py312h68727a3_1    conda-forge
pybind11-global           2.13.5          py312h68727a3_1    conda-forge
pycodestyle               2.12.1             pyhd8ed1ab_0    conda-forge
pycparser                 2.22               pyhd8ed1ab_0    conda-forge
pydata-sphinx-theme       0.15.4             pyhd8ed1ab_0    conda-forge
pydevtool                 0.3.0              pyhd8ed1ab_0    conda-forge
pygments                  2.18.0             pyhd8ed1ab_0    conda-forge
pympler                   1.1                pyhd8ed1ab_0    conda-forge
pyparsing                 3.1.4              pyhd8ed1ab_0    conda-forge
pyproject-metadata        0.8.0              pyhd8ed1ab_0    conda-forge
pyside6                   6.7.2           py312hb5137db_2    conda-forge
pysocks                   1.7.1              pyha2e5f31_6    conda-forge
pytest                    8.3.2              pyhd8ed1ab_0    conda-forge
pytest-cov                5.0.0              pyhd8ed1ab_0    conda-forge
pytest-timeout            2.3.1              pyhd8ed1ab_1    conda-forge
pytest-xdist              3.6.1              pyhd8ed1ab_0    conda-forge
python                    3.12.5          h2ad013b_0_cpython    conda-forge
python-dateutil           2.9.0              pyhd8ed1ab_0    conda-forge
python-fastjsonschema     2.20.0             pyhd8ed1ab_0    conda-forge
python-json-logger        2.0.7              pyhd8ed1ab_0    conda-forge
python_abi                3.12                    5_cp312    conda-forge
pythran                   0.16.1          py312hc39e661_1    conda-forge
pytz                      2024.1             pyhd8ed1ab_0    conda-forge
pyyaml                    6.0.2           py312h66e93f0_1    conda-forge
pyzmq                     26.2.0          py312hbf22597_2    conda-forge
qhull                     2020.2               h434a139_5    conda-forge
qt6-main                  6.7.2                hb12f9c5_5    conda-forge
readline                  8.2                  h8228510_1    conda-forge
referencing               0.35.1             pyhd8ed1ab_0    conda-forge
requests                  2.32.3             pyhd8ed1ab_0    conda-forge
rfc3339-validator         0.1.4              pyhd8ed1ab_0    conda-forge
rfc3986-validator         0.1.1              pyh9f0ad1d_0    conda-forge
rich                      13.7.1             pyhd8ed1ab_0    conda-forge
rich-click                1.8.3              pyhd8ed1ab_0    conda-forge
rpds-py                   0.20.0          py312h12e396e_1    conda-forge
ruff                      0.6.4           py312hd18ad41_0    conda-forge
send2trash                1.8.3              pyh0d859eb_0    conda-forge
setuptools                67.1.0             pyhd8ed1ab_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
sniffio                   1.3.1              pyhd8ed1ab_0    conda-forge
snowballstemmer           2.2.0              pyhd8ed1ab_0    conda-forge
sortedcontainers          2.4.0              pyhd8ed1ab_0    conda-forge
soupsieve                 2.5                pyhd8ed1ab_1    conda-forge
sphinx                    7.4.7              pyhd8ed1ab_0    conda-forge
sphinx-copybutton         0.5.2              pyhd8ed1ab_0    conda-forge
sphinx-design             0.6.1              pyhd8ed1ab_1    conda-forge
sphinxcontrib-applehelp   2.0.0              pyhd8ed1ab_0    conda-forge
sphinxcontrib-devhelp     2.0.0              pyhd8ed1ab_0    conda-forge
sphinxcontrib-htmlhelp    2.1.0              pyhd8ed1ab_0    conda-forge
sphinxcontrib-jsmath      1.0.1              pyhd8ed1ab_0    conda-forge
sphinxcontrib-qthelp      2.0.0              pyhd8ed1ab_0    conda-forge
sphinxcontrib-serializinghtml 1.1.10             pyhd8ed1ab_0    conda-forge
sqlalchemy                2.0.34          py312h66e93f0_0    conda-forge
stack_data                0.6.2              pyhd8ed1ab_0    conda-forge
sysroot_linux-64          2.17                h4a8ded7_16    conda-forge
tabulate                  0.9.0              pyhd8ed1ab_1    conda-forge
terminado                 0.18.1             pyh0d859eb_0    conda-forge
threadpoolctl             3.5.0              pyhc1e730c_0    conda-forge
tinycss2                  1.3.0              pyhd8ed1ab_0    conda-forge
tk                        8.6.13          noxft_h4845f30_101    conda-forge
tokenize-rt               6.0.0              pyhd8ed1ab_0    conda-forge
toml                      0.10.2             pyhd8ed1ab_0    conda-forge
tomli                     2.0.1              pyhd8ed1ab_0    conda-forge
tornado                   6.4.1           py312h66e93f0_1    conda-forge
traitlets                 5.14.3             pyhd8ed1ab_0    conda-forge
types-psutil              5.9.5.6            pyhd8ed1ab_0    conda-forge
types-python-dateutil     2.9.0.20240906     pyhd8ed1ab_0    conda-forge
typing-extensions         4.12.2               hd8ed1ab_0    conda-forge
typing_extensions         4.12.2             pyha770c72_0    conda-forge
typing_utils              0.1.0              pyhd8ed1ab_0    conda-forge
tzdata                    2024a                h8827d51_1    conda-forge
uri-template              1.3.0              pyhd8ed1ab_0    conda-forge
urllib3                   2.2.2              pyhd8ed1ab_1    conda-forge
virtualenv                20.26.4            pyhd8ed1ab_0    conda-forge
wayland                   1.23.1               h3e06ad9_0    conda-forge
wcwidth                   0.2.13             pyhd8ed1ab_0    conda-forge
webcolors                 24.8.0             pyhd8ed1ab_0    conda-forge
webencodings              0.5.1              pyhd8ed1ab_2    conda-forge
websocket-client          1.8.0              pyhd8ed1ab_0    conda-forge
wheel                     0.44.0             pyhd8ed1ab_0    conda-forge
xcb-util                  0.4.1                hb711507_2    conda-forge
xcb-util-cursor           0.1.4                h4ab18f5_2    conda-forge
xcb-util-image            0.4.0                hb711507_2    conda-forge
xcb-util-keysyms          0.4.1                hb711507_0    conda-forge
xcb-util-renderutil       0.3.10               hb711507_0    conda-forge
xcb-util-wm               0.4.2                hb711507_0    conda-forge
xkeyboard-config          2.42                 h4ab18f5_0    conda-forge
xorg-fixesproto           5.0               h7f98852_1002    conda-forge
xorg-inputproto           2.3.2             h7f98852_1002    conda-forge
xorg-kbproto              1.0.7             h7f98852_1002    conda-forge
xorg-libice               1.1.1                hd590300_0    conda-forge
xorg-libsm                1.2.4                h7391055_0    conda-forge
xorg-libx11               1.8.9                hb711507_1    conda-forge
xorg-libxau               1.0.11               hd590300_0    conda-forge
xorg-libxdmcp             1.1.3                h7f98852_0    conda-forge
xorg-libxext              1.3.4                h0b41bf4_2    conda-forge
xorg-libxfixes            5.0.3             h7f98852_1004    conda-forge
xorg-libxi                1.7.10               h4bc722e_1    conda-forge
xorg-libxrender           0.9.11               hd590300_0    conda-forge
xorg-libxtst              1.2.5                h4bc722e_0    conda-forge
xorg-libxxf86vm           1.1.5                h4bc722e_1    conda-forge
xorg-recordproto          1.14.2            h7f98852_1002    conda-forge
xorg-renderproto          0.11.1            h7f98852_1002    conda-forge
xorg-xextproto            7.3.0             h0b41bf4_1003    conda-forge
xorg-xproto               7.0.31            h7f98852_1007    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
yaml                      0.2.5                h7f98852_2    conda-forge
zeromq                    4.3.5                ha4adb4c_5    conda-forge
zipp                      3.20.1             pyhd8ed1ab_0    conda-forge
zlib                      1.3.1                h4ab18f5_1    conda-forge
zstandard                 0.23.0          py312hef9b889_1    conda-forge
zstd                      1.5.6                ha6fb4c9_0    conda-forge
```

</details>

**Comment by melissawm** [comment]
ping @HaoZeke when you have the time

**Comment by ilayn** [comment]
I don't use the xforge products but it seems like you are using `/home/jschmidt/Devel/scipy_perf` and `/home/jschmidt/mambaforge/envs/scipy-dev/` environments. They might have different versions or different states. Could you isolate one single environment and run everything in that including numpy scipy and the requirements?

**Comment by HaoZeke** [comment]
Thanks for the ping @melissawm, :)

@ilayn is right, from the log you can see that the import error is because at first everything is using the right environment, `scipy-dev`:

```
--- stderr ---
Traceback (most recent call last):
  File "/home/jschmidt/mambaforge/envs/scipy-dev/bin/f2py", line 7, in <module>
    from numpy.f2py.f2py2e import main
  File "/home/jschmidt/mambaforge/envs/scipy-dev/lib/python3.12/site-packages/numpy/f2py/__init__.py", line 14, in <module>
    import subprocess
  File "/home/jschmidt/mambaforge/envs/scipy-dev/lib/python3.12/subprocess.py", line 49, in <module>
    import signal
```

However from here, these are not the right modules and are from another environment:

```
  File "/home/jschmidt/Devel/scipy_perf/scipy/scipy/signal/__init__.py", line 292, in <module>
    from . import _sigtools, windows
  File "/home/jschmidt/Devel/scipy_perf/scipy/scipy/signal/windows/__init__.py", line 42, in <module>
    from ._windows import *
  File "/home/jschmidt/Devel/scipy_perf/scipy/scipy/signal/windows/_windows.py", line 7, in <module>
    from scipy import linalg, special, fft as sp_fft
ModuleNotFoundError: No module named 'scipy'
```

OTOH removing `scipy_perf` from `PYTHONPATH` should probably do the trick? It seems very unlikely that the default build setup allows `signal` from `scipy` to shadow the one in the standard libary.

**Comment by JaRoSchm** [comment]
I don't really understand the problem. `scipy_perf` is not added to my `PYTHONPATH`, I just cloned the `scipy` repo to this folder which is just created for this purpose and contains nothing else. Then I entered `scipy_perf/scipy` and executed `python dev.py build`. I didn't add something to `PYTHONPATH` manually and to validate this I just tried to build scipy in my Downloads folder, where exactly the same happens.

**Comment by JaRoSchm** [comment]
Actually I just found the solution for my problem, but don't understand why this is a problem for building scipy. Indeed I added some completely different folder to my `PYTHONPATH` in `.zshrc` using `export PYTHONPATH=${REPOSITORIES}//:${PYTHONPATH}`. After removing this line, `scipy` builds without a problem. Note that $REPOSITORIES is at a completely different locating from `scipy_perf/scipy`. Addtionally, I was able to build `scipy` with this modification of `PYTHONPATH` in my `.zshrc` in the past.

If you want, I can close this issue as the problem is solved for me. If there is something else to investigate, I'm happy to help!
Thanks for your help.
